src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -0,0 +1,376 @@
+
+        ResetKmSettings();
+
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("labels_only", true);
+        SetInputParam("initial_centroids", std::move(InitialCentroids));
+        SetInputParam("kill_empty_clusters", true);
+        SetInputParam("max_iterations", std::move(iterations));
+
+        mlpackMain();
+
+        arma::mat KillEmptyOutput;
+        KillEmptyOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+        ResetKmSettings();
+
+        //Checking that the results are not identical
+
+        for (int j = 0; j <N ; ++j)
+        {
+            BOOST_CHECK_NE(NormalOutput(j), AllowEmptyOutput(j));
+        }
+        for (int j = 0; j <N ; ++j)
+        {
+            BOOST_CHECK_NE(KillEmptyOutput(j), AllowEmptyOutput(j));
+        }
+
+    }
+
+
+/**
+ * Checking that that size and dimensionality of Final Input File is correct when flag --in_place is specified
+ */
+    BOOST_AUTO_TEST_CASE(KmClusteringResultSizeCheck)
+    {
+        constexpr int N = 10;
+        constexpr int D = 4;
+        int C = 2;
+
+        arma::mat InputData = arma::randu<arma::mat>(N, D);
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("in_place", true);
+
+
+        mlpackMain();
+
+        BOOST_REQUIRE_EQUAL(InputData.n_cols, D+1);
+        BOOST_REQUIRE_EQUAL(InputData.n_rows, N);
+    }
+
+
+/**
+ * Ensuring that absence of Input is checked.
+ */
+    BOOST_AUTO_TEST_CASE(KmNoInputData)
+    {
+        constexpr int N = 10;
+        constexpr int D = 4;
+        int C = 2;
+        arma::mat input = arma::randu<arma::mat>(N, D);
+
+        SetInputParam("clusters", std::move(C));
+
+        Log::Fatal.ignoreInput = true;
+        BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+        Log::Fatal.ignoreInput = false;
+    }
+
+/**
+ * Ensuring that absence of Number of Clusters is checked.
+ */
+    BOOST_AUTO_TEST_CASE(KmClustersNotDefined)
+    {
+        constexpr int N = 10;
+        constexpr int D = 4;
+        arma::mat input = arma::randu<arma::mat>(N, D);
+
+        SetInputParam("input", std::move(input));
+
+        Log::Fatal.ignoreInput = true;
+        BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+        Log::Fatal.ignoreInput = false;
+    }
+/**
+ * Checking that all the algorithms yield same results
+ */
+    BOOST_AUTO_TEST_CASE(AlgorithmsSimilarTest)
+    {
+        constexpr int N = 100;
+        constexpr int D = 4;
+        int C = 5;
+        std::string algo = "naive";
+        arma::mat InputData = arma::randu<arma::mat>(N, D);
+
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("algorithm", std::move(algo));
+
+        mlpackMain();
+
+        arma::mat NaiveOutput;
+        NaiveOutput = std::move(CLI::GetParam<arma::mat>("output"));
```

Comment:
Can you name variables with a lowercase first letter place?  `naiveOutput`, for instance.  This applies for all the code; see the style guide:

https://github.com/mlpack/mlpack/wiki/DesignGuidelines#StyleGuidelines

The automatic style checker is not working right now, but I hope to have it back up by the end of the day (it depends on a server being installed in a rack and coming back up cleanly; I am not sure of the status right now).  That should be helpful in working out the style issues.  Another big one is that the tabbing and spacing doesn't seem to be right; it should be two-space tabs (with space characters not tab characters).  If you can fix these things I'd really appreciate it. :)

Reply:
Sorry for the style issue! I'll keep the style guide in my mind. It's just I've a habit to write code as such. I hope style checker will be back soon and you'll not have to worry too much then :)

Reply:
No worries, everyone writes code differently and that is ok. :)  But for the sake of one project we should try and keep it consistent.  Anyway, the infrastructure should hopefully be back online tomorrow but more likely Wednesday.